test/multidevice.h

```diff
@@ -59,6 +59,11 @@ class MultiDeviceTest : public NVFuserTest {
   std::vector<std::pair<const std::string, std::chrono::time_point<std::chrono::high_resolution_clock>>> times;
 };
 
+class ShardingTest
+    : public MultiDeviceTest,
```

Comment:
It's up to you but you can also declare the class in the `test_multidevice_sharding.cpp` file

Reply:
good idea, it will be more self contained.